crates/sui-framework/packages/sui-framework/sources/transfer.move

```diff
@@ -26,8 +26,23 @@ module sui::transfer {
     /// be constructed in the transaction they are created.
     const ESharedNonNewObject: u64 = 0;
 
+    #[allow(unused_const)]
+    /// Serialization of the object failed.
+    const EBCSSerializationFailure: u64 = 1;
+
+    #[allow(unused_const)]
+    /// The object being received is not of the expected type.
+    const EReceivingObjectTypeMismatch: u64 = 2;
+
+    #[allow(unused_const)]
+    /// Represents both the case where the object does not exist and the case where the object is not
+    /// able to be accessed through the parent that is passed-in.
+    const EUnableToReceiveObject: u64 = 3;
+
+    #[allow(unused_const)]
     /// Shared object operations such as wrapping, freezing, and converting to owned are not allowed.
-    const ESharedObjectOperationNotSupported: u64 = 1;
```

Comment:
NB: had to change the value of this, but it's fine since this hasn't been released yet :)